### PR TITLE
Fetching Data In React: Adds link to lesson referenced and add clarification to assignment

### PR DESCRIPTION
**Pull request by nikitarevenco**
## Because

There was a section where a previous lesson was referenced, but a link to it wasn't provided

## This PR

- The lesson that was referenced now has a direct link to it, and no longer is inside of code block
- For one of the assignment items, added clarificaiton up to which point the user should read, since the linked article also talks about GraphQL and Axios 

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/the_react_ecosystem/fetching_data_in_react.md [hunk 1]
```diff
@@ -232,7 +232,7 @@ You've already tasted the complexity of data fetching in this lesson. Each reque
 
 <div class="lesson-content__panel" markdown="1">
 
-1. Read [Modern API data fetching methods](https://blog.logrocket.com/modern-api-data-fetching-methods-react/) for a brief overview of what was discussed in this lesson.
+1. Read [Modern API data fetching methods](https://blog.logrocket.com/modern-api-data-fetching-methods-react/) until you reach the **Axios** section for a brief overview of what was discussed in this lesson.
```
Comment (MaoShizhong) [inline react/the_react_ecosystem/fetching_data_in_react.md:235]:
```suggestion
1. Read [Modern API data fetching methods](https://blog.logrocket.com/modern-api-data-fetching-methods-react/) until you reach the Axios section for a brief overview of what was discussed in this lesson.
```
Nit: Feel like emboldening `Axios` would be a little out of place here.

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Very sensible changes, I think - thank you.
One tiny nit below but otherwise very nice.

**Comment by nikitarevenco** [comment]
> Very sensible changes, I think - thank you. One tiny nit below but otherwise very nice.

Okay!

**Review by MaoShizhong — APPROVED** [review]
:rocket:

**Pull request merged**